src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJob.java

```diff
@@ -0,0 +1,95 @@
+package org.kiwiproject.dropwizard.util.job;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotBlank;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotNull;
+import static org.kiwiproject.concurrent.Async.doAsync;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.base.CatchingRunnable;
+import org.kiwiproject.base.DefaultEnvironment;
+import org.kiwiproject.base.KiwiEnvironment;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+/**
+ * Sets up a job from a {@link Runnable} that can be monitored through health checks to ensure it is running correctly.
+ *
+ * @param <T> a task that extends {@link Runnable}
+ */
+@Slf4j
+@Getter
```

Comment:
I don't think we want blanket getter methods on everything in this class...do we? Seems like we should only have getters on things a client (e.g. a heart check) should reasonably need)

Reply:
We will do this later...